assignments/assignment03.rb

```diff
@@ -48,3 +96,172 @@
+
+
+
+
+#The output isn't working properly.  I was happy to get it this close though.
+#It was difficult to work through the homework without having the lecture
+#recordings to reference.
+
+
+class BankAccount
+  attr :records
+  def initialize csvFileData
+    @records = csvFileData.map do |line|
+      fields = line.split ','
+      create_transaction fields[0], fields[1], fields[2], fields[3].chop
+    end
+
+    @records = records.sort_by { |line| (line[:date].split"/")[1].to_i}
+  end
+
+  def create_transaction(date, payee, amount, type)
+    {date: date, payee: payee, amount: amount, type: type}
+  end
+
+
+  # - summary:
+  #   - starting balance, total deposits, total withdrawals, ending balance
+  # summary = [{start_bal, end_bal, total_d, total_w}]
+  # access by typing december.records.summary
+  def summary
+    total_w = 0
+    total_d = 0
+    self.records.each do |line|
+      if line[:type].downcase == "withdrawal"
+        total_w += line[:amount].to_f.round(2)
+      elsif line[:type].downcase == "deposit"
+        total_d += line[:amount].to_f.round(2)
+      end
+    end
+
+    end_bal = total_d - total_w
+    {start_bal: 0.00, end_bal: end_bal, total_d: total_d, total_w: total_w }
+
+  end
+
+  def extract_withdrawal
+    withdrawals = []
+    self.records.each do |line|
+      if line[:type] == 'withdrawal'
+        withdrawals << line
+      end
+    end
+    withdrawals
+  end
+
+  def extract_deposit
+    deposits = []
+    self.records.each do |line|
+      if line[:type] == 'deposit'
+        deposits << line
+      end
+    end
+    deposits
+  end
+end
+
+
+#------------------------------------
+#------------------------------------
+#------------------------------------
+#------------------------------------
+
+
+
+
+
+
+def render_html title, &block
+<<HTML
+  <!doctype html>
+  <html>
+  #{block.call}
+  </html>
+HTML
+end
+
+def render_head(title)
+<<HEAD
+  <head>
+  <title>#{title}</title>
+  </head>
+HEAD
+end
+
+def render_body(title, &block)
+<<BODY
+  <body>
+  <h1>#{title}</h1>
+  #{block.call}
+  </body>
+BODY
+end
+
+
+def render_summary(summary)
+<<SUMMARY
+  <h2>Summary</h2>
+  <table>
+  <tr><th>Starting Balance</th?  <td>$ #{summary[:start_bal]}<td></tr>
+  <tr><th>Total Deposits</th?    <td>$ #{summary[:total_d]}<td></tr>
+  <tr><th>Total Withdrawals</th? <td>$ #{summary[:total_w]}<td></tr>
+  <tr><th>Ending Balance</th?    <td>$ #{summary[:end_bal]}<td></tr>
+  </table>
+SUMMARY
+end
+
+def render_withdrawals(title, records)
+<<RECORDS
+
+    <h2>#{title}</h2>
+  <table>
+  #{records.map {|r| render_record r}.join "\n"}
+  </table>
+RECORDS
+end
+
+def render_deposits(title, records)
+<<RECORDS
+
+    <h2>#{title}</h2>
+  <table>
+  #{records.map {|r| render_record r}.join "\n"}
+  </table>
+RECORDS
+end
+
+def render_record(r)
+<<RECORD
+  <tr>
+  <td>#{r[:date]}</td>
+  <td>#{r[:payee]}</td>
+  <td>$ #{r[:amount]}</td>
+  </tr>
+
+RECORD
+end
+
+
+def CreateStatement (csvFilename)
```

Comment:
in idiomatic ruby, method and variable names are snake-cased:
`def create_statement(csv_filename)`